src/Roslyn.Suppressions/Roslyn.Suppressions/SuppressionChecker.cs

```diff
@@ -117,8 +118,7 @@ public bool IsSuppressed(Diagnostic reportedDiagnostic, string settingsKey)
         }
 
         private static bool IsSameFile(Diagnostic diagnostic, SuppressedIssue suppressedIssue) =>
-            diagnostic.Location.SourceTree?.FilePath.EndsWith(suppressedIssue.FilePath, StringComparison.OrdinalIgnoreCase)
```

Comment:
The old version of the code would incorrectly match `aaa\file.txt` with `c:\XXXaaa\file.txt`.
The new `IsServerFileMatch` handles this case correctly.